src/stores/Binding/Store.ts

```diff
@@ -714,6 +732,33 @@ const getInitialState = (
         );
     },
 
+    resetCollectionMetadata: (targetCollections, targetBindingUUIDs) => {
+        set(
+            produce((state: BindingState) => {
+                if (hasLength(targetCollections)) {
+                    resetPartialCollectionMetadata(state, targetCollections);
+                } else if (hasLength(targetBindingUUIDs)) {
+                    const evaluatedCollections = Object.entries(
+                        state.resourceConfigs
+                    )
+                        .filter(([uuid, _resourceConfig]) =>
+                            targetBindingUUIDs.includes(uuid)
+                        )
+                        .map(
+                            ([_uuid, resourceConfig]) =>
+                                resourceConfig.meta.collectionName
+                        );
+
+                    resetPartialCollectionMetadata(state, evaluatedCollections);
+                } else {
+                    resetPartialCollectionMetadata(state);
+                }
```

Comment:
In this scenario I think I would be better to just have a single call of `resetPartialCollectionMetadata`.